src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -92,10 +108,8 @@ private void Enqueue(float value)
                 {
                     // It is possible that two threads both think the buffer is full, but only one get to actually flush it, the other
                     // will eventually enter this code path and potentially calling Flushing on a buffer that is not full, and that's okay too.
-                    lock (_bufferedValues)
-                    {
+                    lock (this) // Lock the counter
```

Comment:
> But having said all this, it is not hard in this case to lock on other things (we have a string and a array that can be used. Just say the word and I will change it.

If locking on the array is easy and safe, let's just do that.  If not, what you have is fine.  Thanks.

Locking on strings can sometimes be safe, but it's also on my this-can-be-dangerous-so-I-generally-avoid-it list.